app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -141,18 +143,49 @@ function ContainerTopologyCtrl($scope, $http, $interval, $location, topologyServ
       return self.dblclick(d);
     });
 
+    added.append("image")
+      .attr("xlink:href", function (d) {
+        switch(d.item.kind) {
+          case 'ContainerManager':
+              return icons[d.item.display_kind];
+            //return 'vendor-'+ d.item.display_kind + ".svg";
+          default:
+            return null;
+        }
+      })
+      .attr("height", function(d) {
+         return 40;
+      })
+      .attr("width", function(d) {
+         return 40;
+      })
+      .attr("y", function(d) {
+        return self.getDimensions(d).y;
+      })
+      .attr("x", function(d) {
+        return self.getDimensions(d).x;
+      })
+      .on("contextmenu", function(d){
+        self.contextMenu(this, d);
+      });
+
     added.append("text")
-        .text(function(d) {
-          return self.getIcon(d);
-        })
-      .attr('class', function(d) {
+      .text(function(d) {
           switch(d.item.kind) {
             case 'ContainerManager':
-              return 'icon '+ d.item.display_kind;
+              return null;
             default:
-              return 'icon';
+              return self.getIcon(d);
           }
-        })
+      })
+      .attr('class', function(d) {
+        switch(d.item.kind) {
+          case 'ContainerManager':
```

Comment:
I am not saying that you are making the code any worse than before. I am just saying that generally we are trying not to query for provider classes, VM type, provider type etc. in the UI code. Actully we have an effort running called "Provider plugability" that has the goal of minimizing the number of such places.